go/iceberg/types_test.go

```diff
@@ -0,0 +1,64 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package iceberg_test
+
+import (
+	"encoding/json"
+	"testing"
+
+	"github.com/apache/iceberg/go/iceberg"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func TestRemainingTypes(t *testing.T) {
```

Comment:
nit: `Remaining` is rather a confusing name. The goal here is to test **all** types, including lists/maps/nested structs.

I think `test_types.py` does a great job in testing JSON & string representations of all types, and I think we should do the same here as well

Reply:
@zeroshade I think that the only item that is currently outstanding. We just need to make sure that all types are extensively tested

Reply:
thanks, I think this should be fine now. We should probably re-open this against https://github.com/apache/iceberg-go

Reply:
i'll go do that

Reply:
@nastra Looks like the `iceberg-go` repo is completely empty, the github UI doesn't allow forking an empty repository so that I can submit a PR. Could someone please commit a LICENSE file or empty README.md or something so that I can fork it to create the PR?

Reply:
I added a README to `iceberg-go`, so hopefully things should work now

Reply:
@nastra I've created https://github.com/apache/iceberg-go/pull/1 :smile: in addition to the existing code here I've added general dev script stuff there.